lib/vsc/mympirun/common.py

```diff
@@ -0,0 +1,333 @@
+    return in_range
+
+
+class SchedKlass(object):
+
+    _sched_for = []  # classname is default added
+    _sched_environ_test = []
+    SCHED_ENVIRON_ID = None
+
+    HIDDEN = False
+
+    # factory methods for Sched. To add a new Sched class just create a new class that extends the cluster class
+    # see http://stackoverflow.com/questions/456672/class-factory-in-python
+    @classmethod
+    def _is_sched_for(cls, name=None):
+        """see if this class can provide support for sched class"""
+        if name is not None:
+            # add class name as default
+            return name in cls._sched_for + [cls.__name__]
+
+        # guess it from environment
+        totest = cls._sched_environ_test
+        if cls.SCHED_ENVIRON_ID is not None:
+            totest.append(cls.SCHED_ENVIRON_ID)
+
+        for envvar in totest:
+            envval = os.environ.get(envvar, None)
+            if not envval:
+                continue
+            else:
+                return True
+
+        return False
+
+
+class MpiKlass(object):
+
+    _mpirun_for = None
+    _mpiscriptname_for = []
+    _mpirun_version = None
+
+    HIDDEN = False
+    RUNTIMEOPTION = None
+
+    # factory methods for MPI
+    @classmethod
+    def _is_mpirun_for(cls, mpirun_path):
+        """
+        Check if this class provides support for active mpirun command.
+
+        @param cls: the class that calls this function
+        @return: True if mpirun is located in $EBROOT*, and if $EBVERSION* value matches version requirement
+        """
+        res = False
+
+        mpiname = cls._mpirun_for
+        if mpiname:
+            LOGGER.debug("Checking whether %s (MPI name: %s) matches %s", cls, mpiname, mpirun_path)
+
+            # first, check whether specified mpirun location is in $EBROOT<NAME>
+            mpiroot, mpiversion = eb_root_version(mpiname)
+            if mpiroot:
+                LOGGER.debug("found mpi root: %s", mpiroot)
+                # try to determine resolved path for both, this may file if we hit a non-existing paths
+                try:
+                    mpirun_path_real = os.path.realpath(mpirun_path)
+                    mpiroot = os.path.realpath(mpiroot)
+                except (IOError, OSError) as err:
+                    LOGGER.debug("Failed to resolve paths %s and %s, ignoring it: %s", mpirun_path, mpiroot, err)
+
+                # only if mpirun location is in $EBROOT* location, we should check the version too
+                if mpirun_path.startswith(mpiroot) or mpirun_path_real.startswith(mpiroot):
+                    LOGGER.debug("%s (real %s) is in subdirectory of %s", mpirun_path, mpirun_path_real, mpiroot)
+
+                    # next, check if version meets requirements (checked via _mpirun_version function)
+
+                    # mympirun is not compatible with OpenMPI version 2.0: this version contains a bug
+                    # see https://github.com/hpcugent/vsc-mympirun/issues/113
```

Comment:
Add some tag (like FIXME or XXX) so we can easily see these things we need to handle.